ctapipe/io/hessioeventsource.py

```diff
@@ -51,6 +51,12 @@ def is_compatible(file_path):
         '''This class should never be chosen in event_source()'''
         return False
 
+    @property
+    def subarray(self):
+        with self.pyhessio.open_hessio(self.input_url) as file:
```

Comment:
this of course will fail if the file is a stream. Do we want to break that idea? I think you can change the logic to still work with streams.

Reply:
Its done this way because of the limitation of the hessio library. Do we have to account for streams with hessio? 

pyeventio allows the SubarrayDescription to be generated before starting the event loop.